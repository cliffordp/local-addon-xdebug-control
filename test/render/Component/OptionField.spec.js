const React = require( 'react' )
const Component = require( 'react' ).Component
const renderer = require( 'react-test-renderer' )

const context = {
	'React': React,
}

const OptionField = require( './../../../src/Component/OptionField' )( context )

describe( '<OptionField/>', function () {
	const defaultProps = {
		value: 'foo',
		default: 'foo',
		options: [{value: 'foo', label: 'foo'}, {value: 'bar', label: 'bar'}],
		title: 'Foo',
	}

	it( 'renders correctly with props', function () {
		const component = renderer.create( <OptionField {...defaultProps}/> )
		let tree = component.toJSON()

		expect( tree ).toMatchSnapshot()
	} )
} )
