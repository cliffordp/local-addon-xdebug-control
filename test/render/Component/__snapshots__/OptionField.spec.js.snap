// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OptionField/> renders correctly with props 1`] = `
<li
  className="TableListRow OptionField"
>
  <strong>
    <span
      className="OptionField__Title"
    >
      Foo
    </span>
  </strong>
  <div
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <select
      className="OptionField__Options"
      defaultValue="foo"
      id={undefined}
      name={undefined}
    >
      <option
        value="foo"
      >
        foo
      </option>
      <option
        value="bar"
      >
        bar
      </option>
    </select>
  </div>
</li>
`;
